### ISSUE TITLE
Infinite Helm Release History

### ISSUE THREAD

**Issue by boxboatmatt** [rancher/fleet#232]
I am seeing infinite helm release history with my fleet install.

Somewhat related to rancher/fleet#233 but they appear to be different bugs. The MaxHistory 5 setting does not appear to be honored

**Comment by StrongMonkey** [comment]
@boxboatmatt I can't reproduce this. I deleted the gitRepo(then bundle) and helm release history(stored as secrets in k8s) is deleted. Doesn't uninstall delete the history? Do you have extra steps to reproduce this?

**Comment by StrongMonkey** [comment]
Also what does your gitrepo(manifest) look like? Is it raw k8s manifest, helm chart or kustomize?

**Comment by boxboatmatt** [comment]
This particular bug isn't about deleting the bundle  - #233 covers that issue. This bug covers the issue where all of my fleet bundles retain all of the release history for each helm upgrade. I had 1000s of k8s secrets - rather than 5 per bundle

**Comment by StrongMonkey** [comment]
@boxboatmatt Sorry commenting on the wrong issue, I will take a look at why setting history to 5 does not take effect.

**Comment by StrongMonkey** [comment]
On the side note can you provide some steps to reproduce #233?

**Comment by StrongMonkey** [comment]
@boxboatmatt Do you want to give this a try https://github.com/rancher/fleet/pull/243? Currently I don't think maxHistory can be set.

**Comment by boxboatmatt** [comment]
I would try it out but there doesn't seem to be a way to permanently override the `fleet-agent` deployment image... It keeps reverting to `v0.3.0`

**Comment by janeczku** [comment]
This is easily reproducable:

1. Patch the fleet-agent deployment resource in the downstream cluster using `kubectl edit`, e.g. adding an environment variable to the pod spec
This causes the fleet-agent bundle to enter into a flapping state between `modified` and `waitApplied`, resulting in approx. 15 Helm releases each minute.
2. Watch the number of helm releases growing to infinity (eventually killing the cluster once the etcd db is filled)

```
> helm -n fleet-system list
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                                                                                   APP VERSION
fleet-agent     fleet-system    3685             2021-02-05 18:05:24.368763503 +0000 UTC deployed        fleet-agent-v0.0.0+s-ad50b23019cb9d1ba3ba24283509ec06d3d237fa416bd74ac60c1b69809cb                 
> kubectl get secret -n fleet-system|wc -l
3694
> 
```

**Comment by StrongMonkey** [comment]
@Ellenqs Can you make sure your tested @janeczku 's workflow and see if the latest rc fixed it?

**Comment by Ellenqs** [comment]
The issue is reproduced on rancher `v2.5.5`.
Fleet version: `v0.3.3`

**Steps:**

- Deploy rancher single install.
- Create a downstream cluster.
- Go to system project, find the workload fleet-agent, edit it by adding an environment variable.
- Go to kubectl shell, run command `kubectl get secret -n fleet-system`

**Results**
1. The number of helm release secret is greater than 10 and growing infinitely. 

<img width="318" alt="Screen Shot 2021-02-05 at 5 22 23 PM" src="https://user-images.githubusercontent.com/75277058/107104582-23221d80-67d7-11eb-8fb6-04482a9bb504.png">


<img width="323" alt="Screen Shot 2021-02-05 at 5 22 40 PM" src="https://user-images.githubusercontent.com/75277058/107104563-0b4a9980-67d7-11eb-96a3-efcd3ac89aae.png">

2. The fleet-agent bundle is in a flapping state between `modified` and `waitApplied`.

![Screen Shot 2021-02-06 at 10 40 14 AM](https://user-images.githubusercontent.com/75277058/107126960-c884d200-6867-11eb-87fd-6bd03ede7a17.png)
<img width="386" alt="Screen Shot 2021-02-06 at 10 41 28 AM" src="https://user-images.githubusercontent.com/75277058/107126976-ea7e5480-6867-11eb-9b36-250fabc62178.png">

**Comment by Ellenqs** [comment]
> @Ellenqs Can you make sure your tested @janeczku 's workflow and see if the latest rc fixed it?

The bug fix is validated on rancher `v2.5.6-rc4`.
Fleet version: `v0.3.4-rc3`

**Steps:**

- Deploy rancher single install.
- Create a downstream cluster.
- Go to system project, find the workload fleet-agent, edit it by adding an environment variable.
- Go to kubectl shell, run command `kubectl get secret -n fleet-system`

**Results**

1. The number of helm releases is 10 and is not growing as expected.
<img width="662" alt="Screen Shot 2021-02-05 at 4 31 21 PM" src="https://user-images.githubusercontent.com/75277058/107102602-d5ee7d80-67cf-11eb-86c5-2591c6157d95.png">

<img width="653" alt="Screen Shot 2021-02-05 at 5 16 42 PM" src="https://user-images.githubusercontent.com/75277058/107104359-28cb3380-67d6-11eb-9ef0-1eae969bd38a.png">

The issue of "infinite helm release history" is fixed, so close this one.

2. The fleet-agent bundle state still switch between `modified` and `waitApplied` state. Log a new issue for it: https://github.com/rancher/fleet/issues/267
